src/uct/cuda/cuda_ipc/cuda_ipc_iface.c

```diff
@@ -203,11 +203,6 @@ uct_cuda_ipc_progress_event_q(uct_cuda_ipc_iface_t *iface,
             uct_invoke_completion(cuda_ipc_event->comp, UCS_OK);
         }
 
-        status = iface->unmap_memhandle(cuda_ipc_event->mapped_addr);
-        if (status != UCS_OK) {
-            ucs_fatal("failed to unmap addr:%p", cuda_ipc_event->mapped_addr);
-        }
-
```

Comment:
@bureddy by removing this call in progress path, is unmap still handled in case cudaipc cache is turned on?

cc @pentschev
Did you already verify if the patch works when cache is turned on?

Reply:
This PR won't mess with the unmapping when caching is on. The mechanism for memory mapping is handled separately with and without cache, the change here will basically assign the function pointer to the key object when it's mapped by the uncached function.